core/src/test/java/com/datastax/oss/driver/internal/core/pool/ChannelPoolReconnectTest.java

```diff
@@ -133,4 +136,59 @@ public void should_reconnect_when_channel_starts_graceful_shutdown() throws Exce
 
     factoryHelper.verifyNoMoreCalls();
   }
+
+  @Test
+  public void should_let_current_attempt_complete_when_reconnecting_now()
```

Comment:
@tolbertam Do you remember the integration test that led to the ticket?
I think this unit test covers it, but it would be good to double check.